metrics.go

```diff
@@ -0,0 +1,416 @@
+package sentry
+
+import (
+	"fmt"
+	"hash/crc32"
+	"math"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+type (
+	NumberOrString interface {
+		int | string
+	}
+
+	void struct{}
+)
+
+var (
+	member void
+)
+
+type MetricUnit struct {
+	unit string
+}
+
+func (m MetricUnit) toString() string {
+	return m.unit
+}
+
+func MicroSecond() MetricUnit {
+	return MetricUnit{
+		"microsecond",
+	}
+}
+
+func MilliSecond() MetricUnit {
+	return MetricUnit{
+		"millisecond",
+	}
+}
+
+func Second() MetricUnit {
+	return MetricUnit{
+		"second",
+	}
+}
+
+func Minute() MetricUnit {
+	return MetricUnit{
+		"minute",
+	}
+}
+
+func Hour() MetricUnit {
+	return MetricUnit{
+		"hour",
+	}
+}
+
+func Day() MetricUnit {
+	return MetricUnit{
+		"day",
+	}
+}
+
+func Week() MetricUnit {
+	return MetricUnit{
+		"week",
+	}
+}
+
+func Bit() MetricUnit {
+	return MetricUnit{
+		"bit",
+	}
+}
+
+func Byte() MetricUnit {
+	return MetricUnit{
+		"byte",
+	}
+}
+
+func KiloByte() MetricUnit {
+	return MetricUnit{
+		"kilobyte",
+	}
+}
+
+func KibiByte() MetricUnit {
+	return MetricUnit{
+		"kibibyte",
+	}
+}
+
+func MegaByte() MetricUnit {
+	return MetricUnit{
+		"megabyte",
+	}
+}
+
+func MebiByte() MetricUnit {
+	return MetricUnit{
+		"mebibyte",
+	}
+}
+
+func GigaByte() MetricUnit {
+	return MetricUnit{
+		"gigabyte",
+	}
+}
+
+func GibiByte() MetricUnit {
+	return MetricUnit{
+		"gibibyte",
+	}
+}
+
+func TeraByte() MetricUnit {
+	return MetricUnit{
+		"terabyte",
+	}
+}
+
+func TebiByte() MetricUnit {
+	return MetricUnit{
+		"tebibyte",
+	}
+}
+
+func PetaByte() MetricUnit {
+	return MetricUnit{
+		"petabyte",
+	}
+}
+
+func PebiByte() MetricUnit {
+	return MetricUnit{
+		"pebibyte",
+	}
+}
+
+func ExaByte() MetricUnit {
+	return MetricUnit{
+		"exabyte",
+	}
+}
+
+func ExbiByte() MetricUnit {
+	return MetricUnit{
+		"exbibyte",
+	}
+}
+
+func Ratio() MetricUnit {
+	return MetricUnit{
+		"ratio",
+	}
+}
+
+func Percent() MetricUnit {
+	return MetricUnit{
+		"percent",
+	}
+}
+
+func CustomUnit(unit string) MetricUnit {
+	re := regexp.MustCompile(`[^a-z]+`)
+	return MetricUnit{
+		re.ReplaceAllString(unit, ""),
+	}
+}
+
+type Metric interface {
+	GetType() string
+	GetTags() map[string]string
+	GetKey() string
+	GetUnit() string
+	GetTimestamp() int
```

Comment:
Why does timestamp is in `int` instead of `int64`? In Go, `int` does not always refers to `int64`, on certain machines it's `int32`. Should also add more declaration documentation of what this is: unix second? unix millisecond? unix microsecond?